scripts/qa_crowbarsetup.sh

```diff
@@ -3069,22 +3089,34 @@ function onadmin_rebootcloud()
     pre_hook $FUNCNAME
     get_novacontroller
 
-    local cmachines=$(get_all_discovered_nodes)
+    local cmachines
     local m
+
+    if [[ $hacloud = 1 ]] ; then
+        cluster_node_assignment
+
+        for c in data network services; do
+            local clusternodes_var=$(echo clusternodes${clusternameservices})
+            for m in ${!clusternodes_var}; do
+                power_cycle_and_wait $m
+                m_hostname=$(echo $m | cut -d '.' -f 1)
+                wait_for 400 5 "crowbar node_state status | grep $m_hostname | grep -qiE \"ready$|problem$\"" "node $m_hostname to be online"
+            done
+            complain_if_problem
+        done
+
+        cmachines=$unclustered_nodes
```

Comment:
> But cmachines is not compute nodes. It's just "nodes that are not in any cluster" (and in the non-HA case, that is "all nodes").

Oh yeah, right.  Which kind of proves my point that it needs to be renamed to something less cryptic :-)  What was the `c` in `cmachines` referring to?

So something like `non_cluster_nodes` is appropriate.  In fact you probably don't even need an extra variable; you could just do

    unclustered_nodes=$(get_all_discovered_nodes)

in the non-HA case.

> Also, there's no code here to calculate the value of cmachines; this is done in cluster_node_assignment.

No it's not.  `cluster_node_assignment` calculates the value of `$unclustered_nodes`, but it doesn't set `cmachines`.

> So I'm a bit confused by your comment :-)

The line this discussion is commenting on is:

    cmachines=$unclustered_nodes

and then it's immediately followed by a different assignment to `cmachines` in the `else` clause.  My point is that those assignments should not be coupled to the code which reboots clustered nodes, since the variable has nothing to do with clustered nodes.  IIUC your original point was that rebooting of non-clustered nodes could not be extracted into a separate function because `cmachines` is local to this function, but I can't see any reason why the assignment to `cmachine` can't be split off into `onadmin_reboot_controllers`.